src/Compilers/CSharp/Portable/Symbols/ArrayTypeSymbol.cs

```diff
@@ -408,5 +419,85 @@ public override TResult Accept<TResult>(SymbolVisitor<TResult> visitor)
         }
 
         #endregion
+
+        /// <summary>
+        /// Represents SZARRAY - zero-based one-dimensional array 
+        /// </summary>
+        private sealed class SZArray : ArrayTypeSymbol
```

Comment:
Memory consumption. Inheritance allows us to avoid allocating extra bool per instance and there are other data that are relevant only for one array kind, but not the other.